packages/core/src/codewhispererChat/controllers/chat/messenger/messenger.ts

```diff
@@ -88,6 +90,24 @@ export class Messenger {
             )
         )
     }
+
+    public async countTotalNumberOfCodeBlocks(message: string): Promise<number> {
+        if (message === undefined) {
+            return 0
+        }
+
+        // // To Convert Markdown text to HTML using marked library
+        const html = await marked(message)
+
+        const dom = new JSDOM(html)
+        const document = dom.window.document
+
+        // Search for <pre> elements containing <code> elements
+        const codeBlocks = document.querySelectorAll('pre > code')
```

Comment:
Please test this with some messages that already have content that looks like a proper html snippet that contains `<pre>` and within it nested `<code>`. 

In other words you might need to be more specific here and possibly provide a bit longer chain of descendent (preceding `pre` tag) to make sure not to pick out some part of the message that would turn into a similar looking element. 
The `<code>` within `<pre>` comes from UI and if we use marked library separately, it is not clear how the things will get rendered and if you can have additional metadata (such as class/id names etc.) to increase the probability of positive identification.

Are there any unit tests for this?